### PR TITLE
Update release docs with more info

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Adds more detailed information to the release process.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: devel/release.md [hunk 1]
```diff
@@ -15,7 +15,17 @@ report issues. Typically, there is a single alpha or beta release, but if there
 are a higher than expected number of issues there can be multiple releases
 (e.g, a2 or b2).
 
-## Create or update release branch
+## 1. Update submodules
+
+Update submodules by running the following script:
```
Comment (roycaihw) [inline devel/release.md:20]: could you link to https://github.com/kubernetes-client/python/blob/master/devel/submodules.md#update-submodule instead? In that way we make sure the instructions don't diverge in the future

File: devel/release.md [hunk 2]
```diff
@@ -55,17 +65,24 @@ command:
...
 apply the manual fixes.***
 
 ```bash
 git push upstream $RELEASE_BRANCH
 ```
 
-## Hot issues
+## 4. Hot issues
+
+You can use the `scripts/apply-hotfixes.sh` script to apply the fixes below in
```
Comment (roycaihw) [inline devel/release.md:84]: please add a note that the manual fix steps below are deprecated and exist only for documentation purpose.

File: devel/release.md (hunk 1 above)
Comment (scottilee) [reply devel/release.md:20]: Fixed.

File: devel/release.md (hunk 2 above)
Comment (scottilee) [reply devel/release.md:84]: Fixed.

**Comment by scottilee** [comment]
Addressed comments.

**Comment by roycaihw** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**